url4aac7e6f55_simon0191_tetris3d_tgz-pJ8-Tetris3dJ8/src/vsdk/toolkit/media/RGBAImage.java

```
@@ -0,0 +1,531 @@
+//#endif
+
+    }
+    
+    /** Returns a copy of current image in its own memory */
+    public RGBAImage clone()
+    {
+        RGBAImage copy;
+        int xSize = getXSize();
+        int ySize = getYSize();
+        int x, y;
+
+        copy = new RGBAImage();
+        copy.init(xSize, ySize);
+        for ( x = 0; x < xSize; x++ ) {
+            for ( y = 0; y < ySize; y++ ) {
+                copy.putPixel(x, y, getPixel(x, y));
+            }
+        }
+        return copy;
+    }
+
+    /** Returns a copy of current image in its own memory */
+    public RGBImage exportToRgbImage()
+    {
+        RGBImage copy;
+        int xSize = getXSize();
+        int ySize = getYSize();
+        int x, y;
+        RGBAPixel source;
+        RGBPixel target = new RGBPixel();
+
+        copy = new RGBImage();
+        copy.init(xSize, ySize);
+        for ( x = 0; x < xSize; x++ ) {
+            for ( y = 0; y < ySize; y++ ) {
+                source = getPixel(x, y);
+                target.r = source.r;
+                target.g = source.g;
+                target.b = source.b;
+                copy.putPixel(x, y, target);
+            }
+        }
+        return copy;
+    }
+
+//#ifdef WITH_JAVA_DIRECT_BUFFERS
+    private void writeObject(ObjectOutputStream out) throws IOException
+    {
+        try {
+            int x, y;
+
+            PersistenceElement.writeIntBE(out, xSize);
+            PersistenceElement.writeIntBE(out, ySize);
+            byte arr[] = new byte[4];
+
+            data.rewind();
+            for ( y = 0; y < ySize; y++ ) {
+                for ( x = 0; x < xSize; x++ ) {
+                    arr[0] = data.get();
+                    arr[1] = data.get();
+                    arr[2] = data.get();
+                    arr[3] = data.get();
+                    PersistenceElement.writeBytes(out, arr);
+                }
+            }
+        }
+        catch ( Exception e ) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)